app/controllers/dynamic_pages_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class DynamicPagesController < ApplicationController
+  before_action :require_admin, except: [:index, :show]
+  def index
+    if logged_in?
+      @current_user_is_admin = is_admin?
+      @current_user_is_teacher = is_teacher?
+      @current_user_is_verified_teacher = is_verified_teacher?
+    end
+    @all_dynamic_pages = DynamicPage.all
+  end
+  def delete
+    DynamicPage.destroy(params[:id])
+    redirect_to dynamic_pages_path
+  end
+  def new
+    @dynamic_page = DynamicPage.new(flash[:dynamic_page])
+  end
+  def create
+    @dynamic_page = DynamicPage.new(dynamic_page_params)
+    if DynamicPage.find_by(slug: @dynamic_page.slug)
```

Comment:
Same here, I don't think we need this checking if we move the slug uniqueness checks to the DB.
We should also explicitly set creator_id and `last_editor` in this method, too.